src/transaction-runner.coffee

```diff
@@ -112,6 +104,16 @@ class TransactionRunner
   executeTransaction: (transaction, callback) =>
     configuration = @configuration
 
+    # Add length of body if no Content-Length present
+    # Doing here instead of in configureTransaction, because request body can be edited in before hook
+
+    caseInsensitiveMap = {}
+    for key, value of transaction.request.headers
+      caseInsensitiveMap[key.toLowerCase()] = key
+
+    if not caseInsensitiveMap['content-length'] and transaction.request['body'] != ''
+      transaction.request.headers['Content-Length'] = transaction.request['body'].length
```

Comment:
Can we be assured that `transaction.request['body']` will always be only String, without any multi-byte chars?

I'd say this will fail when used together with `utf-8` characters.

easily fixed with `Buffer.byteLength(transaction.request['body'], 'utf8')`